### PR TITLE
Python: Make types required

### PR DESCRIPTION
**Pull request by Fokko**
image

### CODE REVIEW

File: python/pyiceberg/io/__init__.py [hunk 1]
```diff
@@ -91,11 +96,13 @@ def close(self) -> None:
         ...
 
     @abstractmethod
-    def __enter__(self):
+    def __enter__(self) -> BufferedReader:
```
Comment (rdblue) [inline python/pyiceberg/io/__init__.py:99]: Isn't this an open `OutputStream`?

File: python/pyiceberg/expressions/visitors.py [hunk 2]
```diff
@@ -505,7 +505,7 @@ class _ManifestEvalVisitor(BoundBooleanExpressionVisitor[bool]):
     partition_fields: List[PartitionFieldSummary]
     partition_filter: BooleanExpression
 
-    def __init__(self, partition_struct_schema: Schema, partition_filter: BooleanExpression, case_sensitive):
+    def __init__(self, partition_struct_schema: Schema, partition_filter: BooleanExpression, case_sensitive: bool) -> None:
```
Comment (rdblue) [inline python/pyiceberg/expressions/visitors.py:508]: Doesn't this return `_ManifestEvalVisitor`?

File: python/pyiceberg/io/__init__.py (hunk 1 above)
Comment (Fokko) [reply python/pyiceberg/io/__init__.py]: You're right!

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @Fokko!

**Pull request merged**